### ISSUE TITLE
Double input

### ISSUE THREAD

**Issue by SebDominguez** [raphaelyancey/pyKY040#4]
I'm opening a new issue to hopefully get more visibility there...

Having a similar same issue:
`inc_callback`, `dec_callback`& `chg_callback` get called multiple time for each "steps" of my rotary encoder. `sw_callback` is called once as expected. I tried to change the polling_interval, but this seems to make things worst as if I keep turning the encoder clockwise I'll get the counterclockwise callback executed.

![ezgif com-video-to-gif](https://user-images.githubusercontent.com/52816569/70073357-000b4500-15f9-11ea-81bb-da5b172afc02.gif)

Adding `logger.info(self.polling_interval)` on `pyky040.py` at line 47 and 133 give me the following output:`INFO:root:5`

Maybe this is due to the cheap construction of the rotary encoder module? I'm using those one (I thing you are using the same one)

![iu-1](https://user-images.githubusercontent.com/52816569/70072436-495a9500-15f7-11ea-9bc5-22bef42c9145.jpeg)

With the threaded example the callback get called only twice

![2](https://user-images.githubusercontent.com/52816569/70074451-03073500-15fb-11ea-843f-49d629d1562e.gif)

this is the code I run

```python

# Import the module and threading
from pyky040 import pyky040
import threading
import time

# Define your callback
def up_callback(scale_position):
    print("up")

def down_callback(scale_position):
    print("down")

def pressed_callback():
    print("pressed")

# Init the encoder pins
encoder = pyky040.Encoder(CLK=5, DT=6, SW=13, polling_interval=5)

# Setup the options and callbacks (see documentation)
encoder.setup(scale_min=0, scale_max=100, step=1, inc_callback=up_callback, dec_callback=down_callback, sw_callback=pressed_callback)

# Create the thread
my_thread = threading.Thread(target=encoder.watch)

my_thread.daemon=True

# Launch the thread
my_thread.start()

# Do other stuff
print('Other stuff...')
while True:
    print('Looped stuff...')
    time.sleep(10)
# ... this is also where you can setup other encoders!


```
 
---

Well I just received another rotary encoder (PEC16-4220F-S0024) and I still have this issue

_Originally posted by @SebDominguez in https://github.com/raphaelyancey/pyKY040/issues/1#issuecomment-561257953_

**Comment by raphaelyancey** [comment]
I'm currently working on a rewrite of the library using Linux device tree. It's got an instant response and doesn't miss anything anymore.

Downsides are that there will be a breaking change in the API, and that the encoders must be setup on the Pi before being available in Python scripts (i.e. they would have state).

Take a look at the `evdev` branch for a very early idea.

**Comment by raphaelyancey** [comment]
@SebDominguez I just tested the code, and it seems it's your pins choice that is causing trouble. Try with `CLK=26, DT=4, SW=21` and tell me what you get. If it's also working fine with these pins,  maybe it's some kind of interference between neighbor pins?

**Comment by SebDominguez** [comment]
Ok I'll try this ASAP. I choose those pins because I saw you also used them on your other project. I assumed they was safe to use then.

**Comment by raphaelyancey** [comment]
You're right it did work on the radio, but somehow I also got erratic behavior with these pins on a RPi 3B with Raspbian Buster. Maybe something has changed (kernel, GPIO handling...) since I wrote the tutorial.

**Comment by SebDominguez** [comment]
Yes I also experimented erratic behavior on buster (ghost triggering on GPIO using add_event_detect from RPi.GPIO) I didn't tested on stretch to see if this was caused by the update

**Comment by raphaelyancey** [comment]
If you still experience the issue after changing pins, I updated the library on the [evdev](https://github.com/raphaelyancey/pyKY040/tree/evdev) branch with the new `device` mode. I'd love some feedback when you'll have the time to try it!

**Comment by SebDominguez** [comment]
![ezgif-6-70cd4a3c8746](https://user-images.githubusercontent.com/52816569/72913412-65e49500-3d3d-11ea-9e60-0e5fbee5ed32.gif)

sadly I've tried swapping the pins and also downgrading to stretch  but unfortunately I'm still experimenting the double input from the rotary encoder

**Comment by raphaelyancey** [comment]
Does the output make sense, apart from the fact that the callbacks get executed twice? In other words, do the right callbacks get executed (twice) each time you interact with the encoder?

**Comment by SebDominguez** [comment]
Yes the up callback is executed twice for each clockwise steps and the down callback is executed twice on each counterclockwise steps. At no points a clockwise rotation called the down callback. same for the counterclockwise rotation
